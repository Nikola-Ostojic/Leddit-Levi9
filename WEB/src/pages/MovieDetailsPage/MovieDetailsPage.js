import React, { useEffect, useState } from "react";

import { Button, Container, Grid, LinearProgress } from "@material-ui/core";
import DeleteIcon from "@material-ui/icons/Delete";
import { toast } from "react-toastify";

import { apiCall } from "utils/ApiCall";

import styles from "./MovieDetailsPage.module.css";

const MovieDetailsPage = ({ match, role, history }) => {
  const id = match?.params?.id;
  const [movie, setMovie] = useState(null);
  const [error, setError] = useState(null);

  useEffect(() => {
    apiCall({
      url: `/movies/${id}`,
      method: "GET",
      requiresAuth: true
    })
      .then((result) => {
        const { data } = result;
        setMovie(data);
      })
      .catch((e) => {
        setError(e.message);
      });
  }, [id]);

  if (error) {
    return <div>{error}</div>;
  }

  if (!movie && !error) {
    return <LinearProgress />;
  }

  const deleteMovie = () => {
    apiCall({
      url: `/movies/${id}`,
      method: "DELETE",
      requiresAuth: true
    })
      .then(() => {
        toast.success(`Movie: ${movie.name} deleted!`);
        history.replace("/movies");
      })
      .catch((e) => toast.error("Failed to delete movie"));
  };

  return (
    <Container>
      <Grid container justify="space-between">
        <h1>Title: {movie.name}</h1>
        {role === "Admin" && (
          <Button variant="outlined" onClick={deleteMovie}>
            <DeleteIcon />
          </Button>
        )}
      </Grid>
      <img
        className={styles.image}
        src={movie.imageUrl}
        alt="Failed to load"
      ></img>
    </Container>
  );
};

export default MovieDetailsPage;
