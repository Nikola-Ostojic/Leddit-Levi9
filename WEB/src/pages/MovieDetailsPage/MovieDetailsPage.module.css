.image {
    margin-top: 20px;
    width: 100%;
    object-fit: contain;
}