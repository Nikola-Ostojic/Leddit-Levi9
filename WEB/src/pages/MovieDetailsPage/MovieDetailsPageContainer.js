import { connect } from "react-redux";

import { getUserRole } from "modules/Auth/AuthSelectors";

import MovieDetailsPage from "./MovieDetailsPage";

export const mapStateToProps = (state) => ({
  role: getUserRole(state)
});

export default connect(mapStateToProps)(MovieDetailsPage);
