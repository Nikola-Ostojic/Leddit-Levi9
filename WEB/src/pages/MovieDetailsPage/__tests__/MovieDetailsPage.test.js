import React from "react";
import { shallow } from "enzyme";

import MovieDetailsPage from "../MovieDetailsPage";

describe("MovieDetailsPage tests", () => {
  it("Should match snapshot", () => {
    const wrapper = shallow(<MovieDetailsPage />);

    expect(wrapper).toMatchSnapshot();
  });
});
