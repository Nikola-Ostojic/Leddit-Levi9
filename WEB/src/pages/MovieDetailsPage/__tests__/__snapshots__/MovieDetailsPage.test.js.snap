// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetailsPage tests Should match snapshot 1`] = `<WithStyles(ForwardRef(LinearProgress)) />`;
